src/spikeinterface/core/core_tools.py

```diff
@@ -163,7 +163,8 @@ def make_shared_array(shape, dtype):
     from multiprocessing.shared_memory import SharedMemory
 
     dtype = np.dtype(dtype)
-    nbytes = int(np.prod(shape) * dtype.itemsize)
+    shape = (int(x) for x in shape)  # We need to be sure that shape comes in int and not number scalars
```

Comment:
```suggestion
    shape = tuple(int(x) for x in shape)  # We need to be sure that shape comes in int and not numy scalars
```

Reply:
Cool you saw it :)

Reply:
Yes, thanks, I realized it just now. 

You know this generator notation has fucked me up a couple of times, check this out:

```python
is_one_equal_to_0 = np.all((1 == 0 for _ in range(10)))
if is_one_equal_to_0:
    print("This is great, 1 is equal to 0")
``` 

Syntactic suggar is nice until it is not.

Reply:
That's a cool example.  The tuple/generator for () and (even sometimes I forget)  {} for set in addition to dictionary. Those mess with me. But yeah the accident generator is way too often.